src/org/opensolaris/opengrok/configuration/Project.java

```diff
@@ -26,11 +26,20 @@
 import java.io.File;
 import java.io.FileNotFoundException;
 import java.io.IOException;
+import java.util.Comparator;
 
 /**
  * Placeholder for the information that builds up a project
  */
 public class Project {
+
```

Comment:
if you properly implement equals and hashCode, there is no need for comparator ...